api/src/main/java/org/xnio/ssl/JsseSslConduitEngine.java

```diff
@@ -930,6 +930,22 @@ public void closeOutbound() throws IOException {
         }
     }
 
+    /**
+     * Closes the engine for both inbound and outbound connection, and discards this engine's internal
+     * resources immediately without handshaking if this engine has not been used.
+     *
+     * @throws IOException
+     */
+    void close() throws IOException {
+        if (isFirstHandshake()) {
+            setFlags(WRITE_SHUT_DOWN|READ_SHUT_DOWN);
+            closeEngine();
+        } else {
+            closeInbound();
+            closeOutbound();
```

Comment:
The outbound side still needs to be closed; how about something like this?

```suggestion
            try {
                closeInbound();
            } catch (Throwable t) {
                try {
                    closeOutbound();
                } catch (Throwable t2) {
                    t2.addSuppressed(t);
                    throw t2;
                }
            }
            closeOutbound();
```

Reply:
If I understand correctly, this simplifies to

try {
  closeInbound();
} finally {
  closeOutbound();
}

Reply:
No, because in the `finally` version, if both methods throw an exception, the original exception is lost.  `finally` doesn't handle suppression for you.

Reply:
@dmlloyd  I updated the PR with the suggestion you gave. Also, I see I forgot to add the WRITE_COMPLETE flag to first block of the method.